### PR TITLE
Client build work (HMR, build upgrades, refactoring asset placement)

### PR DESCRIPTION
**Pull request by dannon**
Fixes HMR build to be more reliable, and to work with styles -- you can now edit base.scss (or whatever) and it'll automatically reload an open browser with the new style applied.  

While getting this working I also upgraded a lot of the core devDependencies involved in the process, tweaked our webpack config a bit, and did some general refactoring in terms of build & static artifact placement trying to work towards a single standard 'dist' with less exposed.  I eliminated `june_2007_style` and adjusted the few things touching this.

Style sourcemap resolution should be fixed.

The general production build is marginally faster now.  (but you're really going to want to use the HMR build for developing now, anyway -- it's very fast comparatively).

Addressed comments on a standard for vue v-bind and v-on by enabling vue/strongly-recommended (with prettier compromises asserted) and standardizing the codebase on the recommended practice.

### CODE REVIEW

File: lib/galaxy/web/framework/helpers/__init__.py [hunk 1]
```diff
@@ -71,6 +71,15 @@ def css(*args):
     return stylesheet_link(*urls)
 
 
+def dist_css(*args):
+    """
+    Transition function 'js' helper -- this is the modern way where all bundled
```
Comment (nsoranzo) [inline lib/galaxy/web/framework/helpers/__init__.py]:
```suggestion
    Transition function 'css' helper -- this is the modern way where all bundled
```

File: templates/webapps/tool_shed/repository/common.mako [hunk 2]
```diff
@@ -1002,7 +1002,7 @@
         </td>
     %else:
         <td>
-            <img src="${h.url_for('/static')}/june_2007_style/blue/ok_small.png"/>
+            <img src="${h.url_for('/static')}/ok_small.png"/>
```
Comment (nsoranzo) [inline templates/webapps/tool_shed/repository/common.mako]:
```suggestion
            <img src="${h.url_for('/static')}/style/ok_small.png"/>
```

File: doc/source/admin/galaxy_options.rst [hunk 3]
```diff
@@ -1025,6 +1025,40 @@
...
+
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``mulled_resolution_cache_data_dir``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:Description:
+    Data directory used by beaker for caching mulled resolution
+    requests.
+:Default: ``citations/data``
```
Comment (nsoranzo) [inline doc/source/admin/galaxy_options.rst]: See my comments on https://github.com/galaxyproject/galaxy/pull/8741 from today.

Comment (dannon) [reply doc/source/admin/galaxy_options.rst]: I'll rebase over that when it's fixed in dev.  I don't know what the defaults should be or I'd do it now -- just `mulled/data` `mulled/locks` ?

Comment (nsoranzo) [reply doc/source/admin/galaxy_options.rst]: Neither do I! It was just a heads up.

**Comment by dannon** [comment]
Alright, this should work now, tests have been fixed.

**Comment by bgruening** [comment]
@OlegZharkov can you look at this and try it, please.

**Comment by OlegZharkov** [comment]
> @OlegZharkov can you look at this and try it, please.

Sure. Do you also want me to merge it with https://github.com/galaxyproject/galaxy/pull/9261?

**Comment by bgruening** [comment]
@OlegZharkov no. Just test this. It should be superuseful for you and @dannon appreciates some testing.

**Comment by OlegZharkov** [comment]
@bgruening sure, np.

**Pull request merged**